ExploreASL_ImportMaster.m

```diff
@@ -19,41 +19,77 @@
 
     %% Import Workflow
     
-    % We expect x.DataParPath to be the sourceStructure.json file, so the root directory should be the study directory
-	[x.StudyRoot,~,~] = xASL_fileparts(x.DataParPath);
+    % We expect x.opts.StudyRoot to be the sourceStructure.json file, so the root directory should be the study directory
+    if isfield(x, 'dir')
+        if isempty(x.dir.StudyRoot)
+            [x.dir.StudyRoot,~,~] = xASL_fileparts(x.opts.StudyRoot);
+        end
+    end
 	
     % Check if at least one of the three steps should be performed
-    if sum(x.ImportModules)>0
-        if ~isempty(x.DataParPath)
-            if exist(x.DataParPath,'file') && ~exist(x.DataParPath,'dir')
-                % DICOM TO NII
-                if x.ImportModules(1)==1
-                    xASL_module_Import(x.StudyRoot, x.DataParPath, [], [1 0 0], false, true, false, false, x);
+    missingFields = false; % Fallback
+    if sum(x.opts.ImportModules)>0
+        % DICOM TO NII
+        if x.opts.ImportModules(1)==1
+            if ~isempty(x.dir.sourceStructure)
+                try
+                    xASL_module_Import(x.dir.StudyRoot, x.dir.sourceStructure, x.dir.studyPar, [1 0 0], false, true, false, false, x);
+                catch loggingEntry
+                    fprintf(2,'DICOM to NII module failed...\n');
+                    [x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
```

Comment:
I think we only need the message and the stack. Within the stack there are all related files. If necessary, we only need to change the `xASL_qc_AddLogginInfo` script though and not all the calls if it, since it expects a correct Matlab Exception class object.